Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Window/ShaderManagementConsoleWindow.cpp

```diff
@@ -37,6 +38,42 @@ namespace ShaderManagementConsole
         Base::OnDocumentOpened(documentId);
         m_documentInspector->SetDocumentId(documentId);
     }
+
+    AZStd::string ShaderManagementConsoleWindow::GetSaveDocumentParams(const AZStd::string& initialPath) const
+    {
+        // Get shader file path
+        AZStd::string shaderFullPath;
+        AZ::RPI::ShaderVariantListSourceData shaderVariantList = {};
+        ShaderManagementConsoleDocumentRequestBus::EventResult(
+            shaderVariantList,
+            GetCurrentDocumentId(),
+            &ShaderManagementConsoleDocumentRequestBus::Events::GetShaderVariantListSourceData);
+        shaderFullPath = AZ::RPI::AssetUtils::ResolvePathReference(initialPath, shaderVariantList.m_shaderFilePath);
+        
+        QMessageBox msgBox;
+        msgBox.setText("Where do you want to save the list?");
+        QPushButton* projectBtn = msgBox.addButton(QObject::tr("Save to project"), QMessageBox::YesRole);
+        QPushButton* engineBtn = msgBox.addButton(QObject::tr("Save to engine"), QMessageBox::NoRole);
+        msgBox.exec();
+
+        AZStd::string result;
+        if (msgBox.clickedButton() == projectBtn)
+        {
+            AZ::IO::FixedMaxPath projectPath = AZ::Utils::GetProjectPath();
+            AZStd::string shaderName, shaderPath;
+            AzFramework::StringFunc::Path::Split(shaderFullPath.c_str(), nullptr, &shaderPath, &shaderName);
+            AZStd::string relativePath = shaderPath.substr(shaderPath.find("Assets"), shaderPath.length());
+            AzFramework::StringFunc::Path::StripComponent(relativePath);
+            result = AZStd::string::format("%s\\ShaderVariants\\%s\\%s.shadervariantlist", projectPath.c_str(), relativePath.c_str(), shaderName.c_str());
```

Comment:
You might be able to do this more reliably using one of these functions to get the relative path from the absolute path, add it on to the project folder, and replace the extension.

AssetSystemRequestBus
virtual bool GenerateRelativeSourcePath(const AZStd::string& sourcePath, AZStd::string& relativePath, AZStd::string& rootFilePath) = 0;
virtual bool GetSourceInfoBySourcePath(const char* sourcePath, AZ::Data::AssetInfo& assetInfo, AZStd::string& watchFolder) = 0;